core/src/main/java/com/scalar/db/transaction/consensuscommit/PrepareMutationComposer.java

```diff
@@ -67,7 +67,7 @@ private void add(Put base, @Nullable TransactionResult result) throws ExecutionE
     putBuilder.intValue(Attribute.STATE, TransactionState.PREPARED.get());
     putBuilder.bigIntValue(Attribute.PREPARED_AT, current);
 
-    if (result != null) { // overwrite existing record
+    if (!base.isInsertModeEnabled() && result != null) { // overwrite existing record
```

Comment:
When the insert mode is enabled for a put operation, we always try to write the record of the put operation as an initial record. In other words, even if the target record of the put operation is in the read set, we do not use it for the preparation write when the insert mode is enabled.